src/tensor_product.rs

```diff
@@ -65,14 +65,25 @@ fn tensor_product<Ty: EdgeType>(
                 )
                     .into_py(py),
             );
+        }
+    }
+    if undirected {
+        for edge_first in first.edge_references() {
+            for edge_second in second.edge_references() {
+                let source = hash_nodes
```

Comment:
We should skip self loops here, right? Otherwise we'll add them twice in the output graph.

```suggestion
                if edge_second.source() == edge_second.target() {
                    continue;
                }

                let source = hash_nodes
```

Reply:
Thanks. This do not support multigraph. I fixed it and added its test.